src/rgw/rgw_bucket_logging.cc

```diff
@@ -754,19 +755,19 @@ int get_bucket_id(const std::string& bucket_name, const std::string& tenant_name
   return 0;
 }
 
-int update_bucket_logging_sources(const DoutPrefixProvider* dpp, rgw::sal::Driver* driver, const rgw_bucket& target_bucket_id, const rgw_bucket& src_bucket_id, bool add, optional_yield y) {
+int update_bucket_logging_sources(const DoutPrefixProvider* dpp, rgw::sal::Driver* driver, const rgw_bucket& target_bucket_id, const rgw_bucket& src_bucket_id, const std::string& target_prefix, bool add, optional_yield y) {
   std::unique_ptr<rgw::sal::Bucket> target_bucket;
   const int ret = driver->load_bucket(dpp, target_bucket_id, &target_bucket, y);
   if (ret < 0) {
     ldpp_dout(dpp, 5) << "WARNING: failed to get logging bucket '" << target_bucket_id  <<
       "' in order to update logging sources. ret = " << ret << dendl;
     return ret;
   }
-  return update_bucket_logging_sources(dpp, target_bucket, src_bucket_id, add, y);
+  return update_bucket_logging_sources(dpp, target_bucket, src_bucket_id, target_prefix, add, y);
 }
 
-int update_bucket_logging_sources(const DoutPrefixProvider* dpp, std::unique_ptr<rgw::sal::Bucket>& bucket, const rgw_bucket& src_bucket_id, bool add, optional_yield y) {
-  return retry_raced_bucket_write(dpp, bucket.get(), [dpp, &bucket, &src_bucket_id, add, y] {
+int update_bucket_logging_sources(const DoutPrefixProvider* dpp, std::unique_ptr<rgw::sal::Bucket>& bucket, const rgw_bucket& src_bucket_id, const std::string& target_prefix, bool add, optional_yield y) {
```

Comment:
how is this change related?
isn't that similar to https://github.com/ceph/ceph/pull/66110 ?